common/pipeline_storage/src/main/scala/uk/ac/wellcome/pipeline_storage/Indexer.scala

```diff
@@ -89,17 +88,16 @@ object Indexable extends Logging {
           })
     }
 
-  implicit val workIndexable: Indexable[Work[Identified]] =
-    new Indexable[Work[Identified]] {
+  implicit def workIndexable[State <: WorkState]: Indexable[Work[State]] =
+    new Indexable[Work[State]] {
 
-      def id(work: Work[Identified]) =
-        work.state.canonicalId
+      def id(work: Work[State]): String = work.id
 
-      def version(work: Work[Identified]) =
+      def version(work: Work[State]) =
         work match {
           case Work.Visible(_, _, state) if state.nMergedSources >= 10 =>
             throw new RuntimeException(
-              s"Work ${work.state.canonicalId} has ${work.state.nMergedSources} >= 10 merged sources; versioning/ingest may be inconsistent")
+              s"Work ${work.id} has ${work.state.nMergedSources} >= 10 merged sources; versioning/ingest may be inconsistent")
```

Comment:
I guess this (and adding `nMergedSources`) to `Work[Source]` in the previous file is to allow us to index stuff in intermediate Elasticsearch under the Work IDs?

I’m not sure that’s correct – I was expecting us to use transient IDs (probably UUIDs) for the inter-app messages, possibly combined with something like time-based indices to clean up old documents (https://www.elastic.co/blog/ttl-documents-shield-and-found)

Sidebar: time-based indices might solve the "cleaning up indices in Elasticsearch" issue. If the inter-app message is `(index, ID)`, and the sender writes to a time-based issue, we might be free to use Elastic Cloud.